### ISSUE TITLE
Address failure in TemporaryFolderUsageTest#newFolderWithGivenFolderThrowsIOExceptionWhenFolderCannotBeCreated when run in a Docker image

### ISSUE THREAD

**Issue by cljohnso** [Terracotta-OSS/terracotta-utilities#38]
During tests of #35 and #34 on a broader collection of platforms, when running (CI) in a Docker container, a failure in `org.terracotta.org.junit.rules.TemporaryFolderUsageTest#newFolderWithGivenFolderThrowsIOExceptionWhenFolderCannotBeCreated` is observed (discussion shown below).  This failure needs to be addressed before extending CI to include Docker.

> You can check the results of my `test` branch here: https://github.com/mathieucarbou/terracotta-utilities/actions/runs/1884731142

Interesting -- a failure in a `TemporaryFolder` test in which is appears that `File.setReadOnly()` does not, in fact, set the `File` to read-only is preventing the rest of the tests being run.  An adjustment to the build script needs to be made to allow the remainder of the build to continue following a test failure to ensure my changes are suitable for Docker execution.  As for the apparent `File.setReadOnly()` failure, I'll investigate what the root cause might be.  (The most recent change to `TemporaryFolderUsageTest` was actually in the failing method -- junit-team/junit4#1632.  Our version of this class has that update.)

This failure will need to be addressed, in some way, before moving ahead with the CI expansion.

_Originally posted by @cljohnso in https://github.com/Terracotta-OSS/terracotta-utilities/issues/35#issuecomment-1048841126_

**Comment by cljohnso** [comment]
In addition to currency with JUnit 4, the code underlying the JUnit 5 `TempDir` annotation does not include a similar test so no guidance is available from JUnit 5's approach.

**Comment by cljohnso** [comment]
We'll likely need to examine the operation of the test more closely -- the test result indicates the attempt to call `File.newFolder(...)` against a directory that was altered using `File.setReadOnly()` did not fail as expected.  It's unclear, at this point, if the `newFolder` succeeded or failed in a different way.

```plain
org.terracotta.org.junit.rules.TemporaryFolderUsageTest.newFolderWithGivenFolderThrowsIOExceptionWhenFolderCannotBeCreated  Time elapsed: 0.015 s  <<< FAILURE!
java.lang.AssertionError: Expected test to throw (an instance of java.io.IOException and exception with message a string containing "could not create a folder with the path 'level1'")
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.rules.ExpectedException.failDueToMissingException(ExpectedException.java:263)
	at org.junit.rules.ExpectedException.access$200(ExpectedException.java:106)
	at org.junit.rules.ExpectedException$ExpectedExceptionStatement.evaluate(ExpectedException.java:245)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:377)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:138)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:465)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:451)
```

[TEST-org.terracotta.org.junit.rules.TemporaryFolderUsageTest.xml.zip](https://github.com/Terracotta-OSS/terracotta-utilities/files/8128305/TEST-org.terracotta.org.junit.rules.TemporaryFolderUsageTest.xml.zip)

**Comment by cljohnso** [comment]
Recoding a portion of the `newFolderWithGivenFolderThrowsIOExceptionWhenFolderCannotBeCreated` test as the following allows the test suite (and remainder of the build) to complete with no further errors:

```java
//        thrown.expect(IOException.class);
//        thrown.expectMessage("could not create a folder with the path 'level1'");
        try {
            tempFolder.newFolder("level1");
            throw new AssumptionViolatedException("File creation successful in read-only directory; super-privileged user likely");
        } catch (IOException e) {
            assertThat(e.getMessage(), is("could not create a folder with the path 'level1'"));
        }
```

We may be better off simply _ignoring_ the test altogether -- that would be appropriate if there is no dependency on the `IOException` thrown when a create-in-read-only-directory error is raised in the main code.

**Comment by cljohnso** [comment]
Closed by PR #40
